### ISSUE TITLE
Creating a new Xbox Live Account to use with Xemu

### ISSUE THREAD

**Issue by mitch03** [feudalnate/Original-Xbox-LIVE-Account#3]
Hey, im hoping someone can help me, I would like to create an xbox live account to use with xemu, but im not sure how to link the file with the xemu, any help would be greatly appreciated, thank you!!!

**Comment by CodeAsm** [comment]
There are a few ways, some of them are easier, some of them require some more research by either you or others.
**The live accounts are not files, they are stored within hidden portions of the harddisk or memory cards/units.**. we have chosen a file extension, but microsoft nor xemu the emulator know what to do with those. We need to transfer them onto these secret hidden locations (and in the case on the harddisk location, also do some magic calculation with the matching eeprom). 

Your dashboard/running version of the xbox dash should also be supporting xbox live, else it wont see the live account at all. run a dashupdate if they dont show up and try again. 

Id pick option 1.

1. I guess for most people, https://insignia.live/ might be a easy solution, get some relatively unique eeprom (serial) I think from a original xbox or, if i understand https://xemu.app/docs/eeprom/ correctly, one will be generated for you. When you signup for insignia (and have a live enabled dashboard), they will make the xbox  a live account on your virtual xemu. besides maybe findleing with DNS settings for xemu, youll end up with a live account. looks to be the easiest for Phantasy Star online and maybe online multiplayer if you want.
Someone made what appears a tutorial: https://inputselect.net/xemu-setup-xbox-live-insignia/
and ive spotted some network settings in this video: https://www.youtube.com/watch?v=dKlKEZ8Jf0k
I havent tried these at all yet. but seem doable.

2. use a raw(not cqow2) xbox harddisk image for xemu and use this repo's tool to read/write a new live account to the emulated harddisk. I dont know if writing to the harddisk is supported, if not, try a virtual memory unit. It should be able to write whatever you want Nickname to the memoryunit/harddisk **if this tool does not support writing to harddisk, qcow2 is fine**
   - get xemu working. including a original xbox dashboard that supports xbox live (you may run a older dash and get a dash live updater from a live enabled game to get the live enabled dashboard
   - https://xemu.app/docs/xmus/ follow this.
   - xemu will boot your dashboard and formats the memory unit if you go into the xbox storage menu, it will prompt you and format the XMU (as xemu calls them)
   - when xemu is off, use the repo tool to write any live account to the XMU 
   - transfer Live account form XMU onto the harddisk using the storage menu in the dashboard.
   - profit, fun, awesome. sadly you cannot go online with this anymore, unless its a insignia created account.

3. use a real xbox memory unit or usb storage stick formatted/prepared (or PSP in storage mode was rumoured to work? some smartphones too? rooted im afraid) to use with the OG xbox and connect it to the xemu, how to get a live account onto the memory unit? Either generate one with the tool on this repo, or find an original one and load it in with said program and write to memoryunit/usb stick. (to make a normal usb stick work, insert in a real xbox or connect to xemu and try browse with the dash to the storage menu. it will wipe ALL DATA, and tell you if succeeded.
 Options to get a memorycard in xemu:
   - Real memorycard using a original controller as usb hub+ connector
   - a usb stick with not too much space (4gb or less?) a 512mb one worked for me. USB1.1 should be supported.
   - a raw file acting as a usb memory stick.
  I have mounted read memoryunits onto xemu (and especial back when xqemu still was new) succesfully. something like https://superuser.com/questions/566536/boot-an-existing-gnu-linux-installation-from-a-real-hdd-in-qemu-windows hints how to maybe mount a real harddisk or memory unit to xemu. if you somewhat familiar with qemu, xemu is based on qemu and takes similiar commands, including inside the monitor. Id pick and try memorycards maybe, because if you go harddisk, you may break a real harddisk if doing something wrong or if xemu crashes somehow maybe
   
4. use a eeprom and harddisk image that contain a live account. the eeprom should match the harddisk image. this way, the software in the xbox will use the right keys to load the live account from the harddisk and they may work. if you dont have the matching eeprom, i hope you do have the serial numbers and such to rebuild a seemingly correct eeprom. Someone might be sharing a harddisk image and eeprom with a used to work live account. Dont share insignia accounts this way, it will break eventually in many ways if yall connect at the same time. many cry.

5. manually copy a live account from one memory card to another. this might be "easy" if your familiar with the tools, can be confusing and troublesome if this is all too new for you.
   -  go https://web.archive.org/web/20170627145213/http://assemblergames.com/threads/xbox-live-accounts-xqemu-and-mu.60352/ and read, follow somewhat what i wrote. (sorry if confused)
   - grab one of my (original) live accounts: https://codeasm.com/xbox/live/accounts/ (same as old forum post)
   - use whatever hexeditor or raw byte writing tool you have to get them on your memory card
   - transfer to xemu emulated xbox harddisk (basicly parts of my point 2)
   
TLDR: Go for option 1, its the easiest way to get a live account that even works today. many games that require or like a live account will work fine with this option. and you dont really need to do magic commands or trickery with editors. https://inputselect.net/xemu-setup-xbox-live-insignia/ looks easy enough.
